opengrok-indexer/src/main/java/org/opengrok/indexer/index/IndexDatabase.java

```diff
@@ -486,17 +487,17 @@ public void update() throws IOException {
                     reader.close();
                 }
             }
-
-            try {
```

Comment:
Please don't relocate this. That breaks known failure handling where we would actually not want to finish writing. I.e. `finishWriting()` in advance of a `RuntimeException` catch is deliberate.

Recommend you swallow web exceptions as per your alternative idea. Web exceptions didn't used to abort an index run; you used to be able to index without Opengrok web running. Not sure what changed or when to break that.

Reply:
Oh, okay. Also added a comment there. The API failure is really a soft one.